src/machines/java/com/enderio/machines/common/item/FluidTankItem.java

```diff
@@ -26,4 +49,63 @@ public BlockEntityWithoutLevelRenderer getCustomRenderer() {
             }
         });
     }
+
+    @Override
+    public void appendHoverText(ItemStack stack, @Nullable Level pLevel, List<Component> tooltip, TooltipFlag pFlag) {
+        ItemTooltip.addShiftKeyMessage(tooltip);
+        Optional<IFluidHandlerItem> fluidHandler = stack.getCapability(ForgeCapabilities.FLUID_HANDLER_ITEM).resolve();
+
+        ItemTooltip.addFluidTankMessage(tooltip, fluidHandler.get().getFluidInTank(0));//always present
```

Comment:
not necessarily, just do the ifPresent thing and log warning

Reply:
Done